### DISCUSSION TITLE
casting with selectors on arrays and structs

### DISCUSSION THREAD

**Discussion by lostmygithubaccount** [ibis-project/ibis#6459] [Q&A] [answered]
I have tables with various timestamp columns. I need to make them all consistent, so I'm running something like:

```python
t.mutate(
    s.across(
        s.of_type("timestamp"),
        lambda x: x.cast(dt.Timestamp(timezone=None)),
    )
)
```

I want to apply this to any timestamp column within a struct (or an array of structs). is this possible?

**Comment by cpcloud** [comment]
Is there any reason you need to cast them prior to using them? Why not cast when you access the field?

**Reply by lostmygithubaccount** [reply]
this is the issue I was seeing with writing Delta Lake tables -- I'm able to fix it for most columns, but the cast isn't propagated to structs/arrays of structs

I admit this probably isn't a very relevant use case for Ibis, my next steps is probably to manually cast those few fields remaining causing issues

**Reply by cpcloud** [reply]
I thought we had discovered this was an issue with the deltalake library and not with ibis. Are you trying to make progress without waiting for an upstream fix?

**Reply by lostmygithubaccount** [reply]
yeah and I was further confused after playing around with it more -- I have a workaround now

**Comment by cpcloud** [comment, marked as answer]
Closing this out for now!
